std-bits/table/src/main/java/org/enso/table/data/column/operation/cast/ToBigDecimalConverter.java

```diff
@@ -37,92 +38,92 @@ private Storage<BigDecimal> convertDoubleStorage(
       DoubleStorage doubleStorage, CastProblemAggregator problemAggregator) {
     Context context = Context.getCurrent();
     int n = doubleStorage.size();
-    BigDecimalBuilder builder = new BigDecimalBuilder(n);
+    Builder builder = Builder.getForType(BigDecimalType.INSTANCE, n, problemAggregator);
     for (int i = 0; i < n; i++) {
       if (doubleStorage.isNothing(i)) {
         builder.appendNulls(1);
       } else {
         double x = doubleStorage.getItemAsDouble(i);
         BigDecimal bigDecimal = BigDecimal.valueOf(x);
-        builder.appendRawNoGrow(bigDecimal);
+        builder.appendNoGrow(bigDecimal);
       }
 
       context.safepoint();
     }
-    return builder.seal();
+    return (Storage<BigDecimal>) builder.seal();
```

Comment:
Ideally my preference would be to keep our code free of such casts, as it's much nicer if Java's typesystem can prove this is correct.

I think I'd add a specific `getForBigDecimal` or try to make the `Builder.getForType` return more specific types (the idea is we could make `StorageType` generic over the type it holds and then make `Storage<T> getForType(StorageType<T>)` - this way Java's typesystem 'proves' that the type of `Storage` returned is consistent with the `StorageType` we asked for).

But that's just a nitpick, overall the strategy for Builder.getForType seems like a nice direction so I'm ok with keeping this few casts.

So depends on what you think makes more sense (keeping as is or adding `getForBIgDecimal` or adding some generic types tricks).

Reply:
Have got to revisit this, this is the current workaround to get it compiling.
Want it to be based on the `ColumnStorage` interface but will work on that in next commit once passing all tests again.

Reply:
Ok, I see. Happy to help with abusing the type system, that's essentially my hobby.